pkg/tar/stream.go

```diff
@@ -0,0 +1,151 @@
+package tar
+
+import (
+	"archive/tar"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+)
+
+func StreamFunc(w io.Writer, dir, relativePath string, writeFunc func(f os.FileInfo, shardRelativePath, fullPath string, tw *tar.Writer) error) error {
```

Comment:
copied and adapted from plutonium to be more flexible for the OSS use-case where we filter files before they go to the archive.  the original API is below, and is functionally equivalent to the original.

Reply:
Does this need to be exported?  Also, needs docs.

Reply:
`StreamFunc` is an odd name for a function to call.  Maybe have `Stream` take an optional func arg that defaults to the `TarFile` if nil?

Reply:
👍   needs to be exported, esp since it now replaces Stream(...)